### PR TITLE
[PMD Java] Enable Sider's recommended set of rules

### PR DESCRIPTION
**Pull request by sleeeto**
> Explain a summary, purpose, or background for this change.

Enable Sider's recommended set of rules.

> Refer related issues or pull requests (e.g. `Close #123` or `None`).

https://github.com/sider/sideci/issues/10434

> Check the following if needed.

- [x] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

File: test/smokes/pmd_java/expectations.rb [hunk 1]
```diff
@@ -7,20 +7,9 @@
...
-      location: { start_line: 17, start_column: 16, end_line: 17, end_column: 17 },
-      object: { ruleset: "Performance", priority: "3" },
-      git_blame_info: {
-        commit: :_, line_hash: "846dcdb29796bc0ef3e1c3b6c6ccd7ccd6c352f7", original_line: 17, final_line: 17
-      }
-    },
     {
       message: "Avoid calling finalize() explicitly",
-      links: %W[https://pmd.github.io/pmd-#{default_version}/pmd_rules_java_errorprone.html#avoidcallingfinalize],
+      links: %W[https://pmd.github.io/pmd-6.33.0/pmd_rules_java_errorprone.html#avoidcallingfinalize],
```
Comment (suetsuguk) [inline test/smokes/pmd_java/expectations.rb:12]:
```suggestion
      links: %W[https://pmd.github.io/pmd-#{default_version}/pmd_rules_java_errorprone.html#avoidcallingfinalize],
```

Revert the code, then this test will pass.

File: test/smokes/pmd_java/expectations.rb [hunk 2]
```diff
@@ -29,20 +18,9 @@
...
-      location: { start_line: 20, start_column: 18, end_line: 20, end_column: 29 },
-      object: { ruleset: "Multithreading", priority: "3" },
-      git_blame_info: {
-        commit: :_, line_hash: "928fb2851e5777de1b2700b5b9950bf745e62197", original_line: 20, final_line: 20
-      }
-    },
     {
       message: "Exceptions should not extend java.lang.Error",
-      links: %W[https://pmd.github.io/pmd-#{default_version}/pmd_rules_java_design.html#donotextendjavalangerror],
+      links: %W[https://pmd.github.io/pmd-6.33.0/pmd_rules_java_design.html#donotextendjavalangerror],
```
Comment (suetsuguk) [inline test/smokes/pmd_java/expectations.rb:23]: Same as above.

File: test/smokes/pmd_java/expectations.rb (hunk 1 above)
Comment (sleeeto) [reply test/smokes/pmd_java/expectations.rb]: OK, ready.

File: test/smokes/pmd_java/expectations.rb (hunk 2 above)
Comment (sleeeto) [reply test/smokes/pmd_java/expectations.rb]: OK, fixed.

File: test/smokes/pmd_java/expectations.rb [hunk 3]
```diff
@@ -53,25 +31,14 @@
     },
     {
       message: "Do not use hard coded encryption keys",
-      links: %W[https://pmd.github.io/pmd-#{default_version}/pmd_rules_java_security.html#hardcodedcryptokey],
+      links: %W[https://pmd.github.io/pmd-6.33.0/pmd_rules_java_security.html#hardcodedcryptokey],
```
[the remark this answers is not in the document]
Comment (sleeeto) [reply test/smokes/pmd_java/expectations.rb]: OK, fixed.

**Comment by sleeeto** [comment]
@ybiquitous Ready.

**Review by suetsuguk — APPROVED** [review]
I have verified your changes. Thank you.

**Review by ybiquitous — CHANGES_REQUESTED** [review]
@sleeeto Can you resolve the conflict, please?

**Comment by sleeeto** [comment]
@ybiquitous OK, resolved.

**Review by ybiquitous — APPROVED** [review]
Thank you 👍🏼

**Pull request merged**